src/sig-verifier-js/tests.ts

```diff
@@ -1,18 +1,142 @@
+const II_CANISTER_ID: string = "fgte5-ciaaa-aaaad-aaatq-cai";
+// User identity as viewed by the FE that called the II-instance to get DELEGATION_CHAIN_JSON.
+const USER_PRINCIPAL_ID: string =
+  "hf7wk-a35mp-bc6eb-ntvr2-aeu3d-naglw-n6ea3-qn5ps-jcanu-p2vro-5ae";
+
+// Equal to delegation.pubkey-field in DELEGATION_CHAIN_JSON
+const CHALLENGE = new Uint8Array([
+  0xe7, 0x87, 0x5e, 0x69, 0xce, 0x7b, 0xed, 0xa6, 0xfc, 0x7b, 0x6d, 0xfb, 0xd9,
+  0xb7, 0x5b, 0xe1, 0xc6, 0xf6, 0xd5, 0xde, 0xba, 0xe3, 0xae, 0x1e, 0xd7, 0xc7,
+  0xf8, 0x73, 0xde, 0x1b, 0x6f, 0x9f, 0x75, 0xe9, 0xe7, 0xdc, 0xdd, 0xcf, 0x37,
+  0xef, 0xad, 0xdc, 0xdf, 0x6f, 0x7b, 0x69, 0xa7, 0xb5, 0x73, 0x77, 0xb5, 0xdd,
+  0xae, 0xf8, 0x7d, 0xee, 0x38, 0x6d, 0xdd, 0x75, 0xe3, 0x9e, 0x9c, 0xd3, 0x9d,
+  0x7d, 0x77, 0xde, 0xbc, 0x79, 0xdf, 0x1b, 0x7b, 0x46, 0x9d, 0xf3, 0x6e, 0xb8,
+  0xe7, 0xce, 0xf4, 0x7b, 0x4d, 0x5c, 0xef, 0xa7, 0xf5, 0xdf, 0x67, 0xdb, 0xef,
+  0xc7, 0x3d, 0xeb, 0xdf, 0x5c,
+]);
+
+export var ROOT_PUBLIC_KEY_RAW = new Uint8Array([
+  0x81, 0x4c, 0x0e, 0x6e, 0xc7, 0x1f, 0xab, 0x58, 0x3b, 0x08, 0xbd, 0x81, 0x37,
+  0x3c, 0x25, 0x5c, 0x3c, 0x37, 0x1b, 0x2e, 0x84, 0x86, 0x3c, 0x98, 0xa4, 0xf1,
+  0xe0, 0x8b, 0x74, 0x23, 0x5d, 0x14, 0xfb, 0x5d, 0x9c, 0x0c, 0xd5, 0x46, 0xd9,
+  0x68, 0x5f, 0x91, 0x3a, 0x0c, 0x0b, 0x2c, 0xc5, 0x34, 0x15, 0x83, 0xbf, 0x4b,
+  0x43, 0x92, 0xe4, 0x67, 0xdb, 0x96, 0xd6, 0x5b, 0x9b, 0xb4, 0xcb, 0x71, 0x71,
+  0x12, 0xf8, 0x47, 0x2e, 0x0d, 0x5a, 0x4d, 0x14, 0x50, 0x5f, 0xfd, 0x74, 0x84,
+  0xb0, 0x12, 0x91, 0x09, 0x1c, 0x5f, 0x87, 0xb9, 0x88, 0x83, 0x46, 0x3f, 0x98,
+  0x09, 0x1a, 0x0b, 0xaa, 0xae,
+]);
+
+test("Should validateDelegationAndGetPrincipal", async () => {
+  let principal = await validateDelegationAndGetPrincipal(
+    CHALLENGE,
+    DELEGATION_CHAIN_JSON,
+    EXPIRATION - BigInt(42),
+    II_CANISTER_ID,
+    ROOT_PUBLIC_KEY_RAW
+  );
+  expect(principal).toEqual(USER_PRINCIPAL_ID);
+});
+
+test("Should fail validateDelegationAndGetPrincipal with wrong challenge", async () => {
+  const call = async () => {
+    await validateDelegationAndGetPrincipal(
+      Uint8Array.from([1, 2, 3, 5]), // wrong challenge
+      DELEGATION_CHAIN_JSON,
+      EXPIRATION - BigInt(42),
+      II_CANISTER_ID,
+      ROOT_PUBLIC_KEY_RAW
     );
-  } catch (e) {
-    // Expect an error string generated from the Wasm module
-    expect(e.toString()).toContain(
-      "Error in DER encoding: Bad length field in boolean block: 2"
+  };
+  expect(call).rejects.toThrow(
+    expect.stringContaining("does not match the challenge")
+  );
+});
+
+test("Should fail validateDelegationAndGetPrincipal with wrong delegation chain", async () => {
+  let WRONG_DELEGATION_CHAIN_JSON: string = `{
+    "delegations": [
+      {
+        "delegation":{
+          "expiration":"17b5b384762bfd21",
+          "pubkey":"e7875e69ce7beda6fc7b6d"
+        },
+      }
+    ],
+    "publicKey":"deadbeef"
+  }`; // missing "signature" after "delegation"
+  const call = async () => {
```

Comment:
here and elsewhere you could drop the `async/await`:

`const call = () => validateDelegationAndGetPrincipal(...)`

or even

`const result = validateDelegationAndGetPrincipal(...)`

as far as I'm concerned you can even inline the whole thing in `expect(validateDelegationAndGetPrincipal(...)).etc`

Reply:
removed `async/await`, thanks!